fms/models/llama.py

```diff
@@ -228,11 +226,10 @@ def from_config(cls, config: LLaMAConfig) -> "LLaMA":
         return cls(config)
 
     def reset_params(self):
-        # Modules are self-initializing, we're just going to down-scale the final prediction head to be
-        # mixed-fan (inputs and gradients scale to the same inverse factors) if it isn't tied
-        self.shared.head.weight.data.normal_(
-            0, 1 / math.sqrt(math.sqrt(self.width * self.shared.vocab_size))
-        )
+        # Call reset_params for relevant sub-layers
```

Comment:
does llama tie weights? i.e. should we avoid reseting parameters for `self.shared.head.weight` 2x?

Reply:
The `reset_params` method of the embedding layer should take care of initializing the head, if not tying, or tying it, if not

Reply:
ah for some reason I was thinking there might be two references to it, but of course it will only be present once in `modules()`